src/Type/Php/ImplodeFunctionReturnTypeExtension.php

```diff
@@ -62,6 +64,17 @@ public function getTypeFromFunctionCall(
 			}
 		}
 
+		if ($arrayType->getIterableValueType()->isArray()->yes()) {
+				return new ErrorType();
+		}
+		if ($arrayType->getIterableValueType() instanceof UnionType) {
+			foreach ($arrayType->getIterableValueType()->getTypes() as $subType) {
+				if ($subType->isArray()->yes()) {
+					return new ErrorType();
```

Comment:
This isn't going to result in any rule reporting this as an error. You need to write a new rule.